python-package/lightgbm/dask.py

```diff
@@ -677,15 +941,33 @@ def fit(
         y: _DaskCollection,
         sample_weight: Optional[_DaskCollection] = None,
         init_score: Optional[_DaskCollection] = None,
+        eval_set: Optional[List[Tuple[_DaskCollection, _DaskCollection]]] = None,
+        eval_names: Optional[List[str]] = None,
+        eval_sample_weight: Optional[List[_DaskCollection]] = None,
+        eval_class_weight: Optional[List[Union[dict, str]]] = None,
+        eval_init_score: Optional[List[_DaskCollection]] = None,
+        eval_metric: Optional[Union[Callable, str, List[Union[Callable, str]]]] = None,
+        early_stopping_rounds: Optional[int] = None,
         **kwargs: Any
     ) -> "DaskLGBMClassifier":
         """Docstring is inherited from the lightgbm.LGBMClassifier.fit."""
+        if early_stopping_rounds is not None:
+            raise RuntimeError('early_stopping_rounds is not currently supported in lightgbm.dask')
+
+        if eval_metric:
+            kwargs['eval_metric'] = eval_metric
```

Comment:
similar comments here about aliases. Is there any reason you couldn't just pass `eval_metric=eval_metric` to `_lgb_dask_fit` instead of appending it into `**kwargs`?

Reply:
Yeah so `eval_class_weight` is a fit param - it doesn't need to get distributed into parts - so at *some point* it needs to be wrapped up into `**kwargs` prior to `client.submit(_train_part)`. That can happen either in the dask estimator subclass's `fit`, `_lgb_dask_fit`, or during  `_train`. I figured since this was a ranker-specific arg, to put it into `DaskLGBMRanker`'s own `fit` method. But you're totally right! Seeing now that this is redundant and it makes sense to do this repackaging in `_lgb_dask_fit`.

Will address aliases, sorry hadn't considered them - does `eval_metric` have aliases??

Reply:
no prob, no prob. Yep `metric` can also be `metrics` or `metric_types`: https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric

Reply:
Hey so on this - `metric` (and aliases `metric_types`) are constructor arguments, I think, and not viable `fit` args. I think that `eval_metrics` does not have any aliases.

```
import numpy as np
import pandas as pd
import lightgbm as lgb

X_df = pd.DataFrame(np.random.normal(size=(100, 5)), columns=[f'f_{i}' for i in range(5)])
y_df = pd.Series(np.random.normal(size=(100,)))
metrics = ['l2', 'mape']

# -- validation set with metrics estimator attr
reg = lgb.LGBMRegressor(metrics=metrics)
reg.fit(X_df, y_df, eval_set=[(X_df.iloc[0:30,], y_df[0:30])])
print(f'reg.metrics = {reg.metrics}')


# -- metric/metrics/metric_types as fit args: unexpected keyword
reg = lgb.LGBMRegressor()
reg.fit(X_df, y_df, eval_set=[(X_df.iloc[0:30,], y_df[0:30])], metrics=metrics)
```

Reply:
oh man I'm so sorry, you're totally right. That's my fault, sorry for wasting your time :grimacing:

Reply:
Ha absolutely no sweat